### ISSUE TITLE
cannot validate certificate for because it doesn't contain any ip sans

### ISSUE THREAD

**Issue by hieu29791** [weaveworks/eksctl#1060]
Hi,
I created cluster on AWS with eksctl, using ubuntu image. Everything is fine, but when I type **kubectl logs**  or **helm ls**, error show like this:

> cannot validate certificate for because it doesn't contain any ip sans. I don't know what wrong

Here is my config file:  
```
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig
metadata:
  name: TestCluster
  region: ap-southeast-1
  version: "1.12"
vpc:  
  nat:
    gateway: Single

#IAM of ServiceRole
iam:
  serviceRoleARN: "arn:aws:iam::*******:role/eksServiceRole"

nodeGroups:
  - name: Worker-Nodes-Group
    instanceType: t3.large
    desiredCapacity: 2
    minSize: 2
    maxSize: 10
    volumeSize: 100
    volumeType: gp2
    ami: ami-01168bce413d4be53
    amiFamily: Ubuntu1804
    volumeEncrypted: true
    privateNetworking: true
    ssh:
      publicKeyName: *******
    iam:
      attachPolicyARNs:
        - arn:aws:iam::*******:policy/k8s-asg-policy
        - arn:aws:iam::aws:policy/Amazon*******
        - arn:aws:iam::aws:policy/Amazon*******
        - arn:aws:iam::aws:policy/Amazon*******
      withAddonPolicies:
        autoScaler: true
        albIngress: true
```
I create some deployment/pod,  state is running, seem like ok, no error appear.

**Comment by errordeveloper** [comment]
What command did you run that produced this error message? Can you provide full output please?

**Comment by hieu29791** [comment]
**Kubectl logs -f** .... and **helm ls**, when I type that, error log appear:  
`Error from server: Get https://192.168.0.12:10250/containerLogs/kube-system/container-name: x509: cannot validate certificate for 192.168.0.12 because it doesn't contain any IP SANs.`

**Comment by errordeveloper** [comment]
Could you please provide the output of `kubectl version`?

Also, could you please try creating an Amazon Linux 2 nodegroup instead of Ubuntu and test if you get the same issue with pods running on that nodegroups? It's possible this is somehow Ubuntu-specific.
You can just added another nodegroup definition with a new name in your config and simply omit `ami` & `amiFamily` fields, then run `eksctl create ng -f <path>`. You will get `aws-node` pods on the new nodes right away, so you could just try `kubectl logs` for one of those.

**Comment by hieu29791** [comment]
I tested with some cases:
1. eksctl
Ubuntu image error as above I describe.
AmazonLinux2 working ok.
2. Default way (https://docs.aws.amazon.com/eks/latest/userguide/getting-started-console.html)
Both Ubuntu and AmazonLinux2 working ok.  
So couch you try to create new cluster with ubuntu image, content like I post above. Thanks

**Comment by JanneEN** [comment]
We've had this issue as well, it seems to be related to the fix #994 with the Ubuntu bootstrap script not working properly from 1.11 version onwards.

**Comment by errordeveloper** [comment]
@JanneEN could you please explain how this relates to #994?

**Comment by JanneEN** [comment]
We were running to the error "cannot validate certificate for because it doesn't contain any ip sans" in a cluster installed with eksctl using Ubuntu AMI starting with version 1.11. The same error didn't occur if we installed the cluster using console or CLI and using the AWS provided CloudFormation template for the worker nodes (which is from 1.11 onwards the same template as for Amazon Linux AMI). Currently we are unable to install a supported cluster using eksctl for Ubuntu AMI due to this issue.

The issue referenced in #994 is fixing a problem introduced in Kubernetes 1.11 in #350 which relates to kubelet initialization changes between versions 1.10 and 1.11 that cause the "cannot validate certificate for because it doesn't contain any ip sans" error with specific commands (e.g. helm list, kubectl logs). So it should fix this issue as well if it relates to the same underlying issue.

**Comment by rodrigc** [comment]
@errordeveloper  I can confirm the same problem with eksctl 0.3.1

**Comment by JanneEN** [comment]
@rodrigc For us this issue was fixed in release 0.4.0 that contained the fix for #994.

**Comment by cristian-radu** [comment]
closing
